solutions/src/main/java/ru/sberbank/school/task10/ScalableThreadPool.java

```diff
@@ -0,0 +1,136 @@
+package ru.sberbank.school.task10;
+
+import lombok.NonNull;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+
+public class ScalableThreadPool implements ThreadPool {
+    private final Queue<FutureTask> tasks = new LinkedList<>();
+    private PoolWorker[] threads;
+    private int minSizePool;
+
+    public ScalableThreadPool(int min, int max) {
+        if (min <= 0 || max <= 0) {
+            throw new IllegalArgumentException("Amount of threads must be more than zero");
+        }
+        if (min > max) {
+            throw new IllegalArgumentException("Minimum amount of threads must be less than maximum");
+        }
+
+        threads = new PoolWorker[max];
+
+        for (int i = 0; i < threads.length; i++) {
+            threads[i] = new PoolWorker("ThreadPoolWorker-" + i);
+        }
+
+        minSizePool = min;
+    }
+
+    @Override
+    public void start() {
+        for (int i = 0; i < minSizePool; i++) {
+            threads[i].start();
+        }
+    }
+
+    @Override
+    public void stopNow() {
+        for (PoolWorker thread : threads) {
+            thread.interrupt();
+        }
+
+        threads = null;
+
+        synchronized (tasks) {
+            tasks.clear();
+        }
+    }
+
+    @Override
+    public void execute(@NonNull Runnable runnable) {
+        if(threads == null) {
```

Comment:
WhitespaceAround: &#39;if&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1883059029>*